internal/pkg/templating/controllerTemplate.go

```diff
@@ -0,0 +1,128 @@
+package templating
+
+import (
+	"bytes"
+	"text/template"
+)
+
+// ControllerTemplateConfig contains the necessary fields to render the controller template
+type ControllerTemplateConfig struct {
+	Kind                 string
+	LowerKind            string
+	OwnerAPIVersion      string
+	ImportMap            map[string]string
+	ResourceWatches      []string
+	ResourceForReconcile []string
+}
+
+// NewControllerTemplateConfig returns the necessary templating configuration
+func NewControllerTemplateConfig(kind, lowerKind, ownerAPIVersion string, importMap map[string]string, resourceWatches, resourceForReconcile []string) *ControllerTemplateConfig {
```

Comment:
This function does nothing but return an object. It may make more sense for this function to take the responsibility for parsing and generating the values that are passed into it. For example, `internal/pkg/templating/template.go` currently creates the function arguments inline. Thus, it seems that it would make more sense to have those preliminary parsing functionalities encapsulated within this function.